engine/crates/engine-v2/schema/src/builder/mod.rs

```diff
@@ -27,10 +28,19 @@ use requires::*;
 impl TryFrom<Config> for Schema {
     type Error = BuildError;
 
-    fn try_from(mut config: Config) -> Result<Self, Self::Error> {
-        let mut ctx = BuildContext::new(&mut config);
-        let sources = ExternalDataSources::build(&mut ctx, &mut config);
-        let (graph, introspection) = GraphBuilder::build(&mut ctx, &sources, &mut config)?;
+    fn try_from(config: Config) -> Result<Self, Self::Error> {
+        let graph = federated_graph::from_sdl(&config.graph)?;
+        (config, graph).try_into()
+    }
+}
+
+impl TryFrom<(Config, FederatedGraphV4)> for Schema {
+    type Error = BuildError;
+
+    fn try_from((mut config, mut graph): (Config, FederatedGraphV4)) -> Result<Self, Self::Error> {
```

Comment:
I would not add this method. This optimization is only useful for the CLI which is kind of irrelevant IMHO. Even there the optimization likely doesn't even matter. The type signature doesn't guarantee in any way that `graph` must be from coming from the `config`

Reply:
Removing that.